### PR TITLE
Update deprecated API list

### PR DESCRIPTION
**Pull request by matyas-streamhpc**

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -53,15 +53,16 @@ On non-AMD platforms, like NVIDIA, HIP provides header files required to support
 
 :::{grid-item-card} Reference
 
-* {doc}`/doxygen/html/index`
+* [HIP runtime API](./reference/hip_runtime_api_reference)
+  * {ref}`modules_reference`
+  * {doc}`reference/hip_runtime_api/global_enum_and_defines`
+  * {doc}`reference/hip_runtime_api/data_structures`
+  * {doc}`reference/hip_runtime_api/files`
+* [HSA runtime API for ROCm](./reference/virtual_rocr)
```
Comment (randyh62) [inline docs/index.md]: Why is this here? I don't see virtual_rocr in the file list, and is HSA from ROCR-runtime needed here?

File: docs/reference/hip_runtime_api_reference.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+.. _runtime_api_reference:
+
+*******************************************************************************
+HIP runtime API
+*******************************************************************************
+
+This is the full HIP Runtime API reference. 
+
+:doc:`Modules <hip_runtime_api/modules>`
```
Comment (randyh62) [inline docs/reference/hip_runtime_api_reference.rst]: The structure here seems kind of redundant. We have the Modules listed on the HIP Runtime API page (here), on the Modules List page, and as individual Modules listed below the Modules list. Seems like we could eliminate one of these layers. UI would suggest either the top-level HIP API page mentions and describes the four subcategories, but leaves the listing to those pages, or lists them, and eliminates the sub-pages.

File: docs/reference/hip_runtime_api/modules_list.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+* :ref:`stream_memory_operations_reference`
+* :ref:`event_management_reference`
+* :ref:`memory_management_reference`
+
+  * :ref:`external_resource_interoperability_reference`
+  * :ref:`stream_ordered_memory_allocator_reference`
+  * :ref:`unified_memory_reference`
+  * :ref:`virtual_memory_reference`
+  * :ref:`texture_management_reference`
+  * :ref:`texture_management_deprecated_reference`
```
Comment (randyh62) [inline docs/reference/hip_runtime_api/modules_list.rst]: Do we need to list the deprecated modules here? They will also appear in the Deprecated API list. If the intention with deprecation is to get people to stop using the features, then listing them once under Deprecated APIs should be enough.

File: docs/reference/hip_runtime_api/data_structures.rst [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+.. _data_structures_reference:
+
+*******************************************************************************
+Data structures
+*******************************************************************************
+
+The API is organized into modules based on functionality:
+
+.. include:: data_structures_list.rst
```
Comment (randyh62) [inline docs/reference/hip_runtime_api/data_structures.rst]: It seems like the list could just be added here instead of in a different file?

File: docs/sphinx/_toc.yml.in [hunk 5]
```diff
@@ -37,21 +37,61 @@ subtrees:
...
+          - file: reference/hip_runtime_api/modules/callback_activity_apis
+          - file: reference/hip_runtime_api/modules/graph_management
+          - file: reference/hip_runtime_api/modules/opengl_interoperability
+          - file: reference/hip_runtime_api/modules/cooperative_groups_reference
+      - file: reference/hip_runtime_api/global_enum_and_defines
+      - file: reference/hip_runtime_api/data_structures
+        subtrees:
+        - entries:
+          - file: doxygen/html/annotated
+          - file: doxygen/html/classes
```
Comment (randyh62) [inline docs/sphinx/_toc.yml.in]: Is Classes needed? It seems like it is re-presenting the content from the annotated page? Do we need both views of the same content?

File: docs/sphinx/_toc.yml.in [hunk 6]
```diff
@@ -37,21 +37,61 @@ subtrees:
...
+          - file: doxygen/html/inherits
```
Comment (randyh62) [inline docs/sphinx/_toc.yml.in]: The image on this page is interesting, but hard to see in its current form. Is there some way to make it a clickable image so that the user can view it all?

File: docs/sphinx/_toc.yml.in [hunk 7]
```diff
@@ -37,21 +37,61 @@ subtrees:
...
+          - file: doxygen/html/functions_data_fields
```
Comment (randyh62) [inline docs/sphinx/_toc.yml.in]: I think we discussed removing this Doxygen content from the TOC? I think it is covered in other topics?

File: docs/reference/hip_runtime_api/modules_list.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
+* :ref:`driver_reference`
```
Comment (randyh62) [inline docs/reference/hip_runtime_api/modules_list.rst]: Is there any way to indicate which files these modules can be found in?

File: docs/reference/hip_runtime_api/files.rst [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+.. _files_reference:
+
+*******************************************************************************
+Files
+*******************************************************************************
+
+The list of all files with brief descriptions and a list of all functions,
+variables, defines, enums, and typedefs with links to the files they belong to.
+
+.. include:: files_list.rst
```
Comment (randyh62) [inline docs/reference/hip_runtime_api/files.rst]: It seems like the list could be added here instead of listed in another file? 

Also, a strange thing happens when reviewing the PR readthedocs.
https://advanced-micro-devices-demo--3591.com.readthedocs.build/projects/HIP/en/3591/reference/hip_runtime_api/files.html

If you click the File List link on this page it takes you to the currently published docs (out of the PR). 
If you click on the File link in the TOC, or click the File List at the bottom under Next, it takes you to the File List in the PR.

**Review by randyh62 — COMMENTED** [review]
Left comments. Could use a few tweaks. I like the approach this is taking with regard to the structure, but I think we could pull out more of the autogenerated content if we are going to replace it with hand-crafted content. I also think there is a bit too much redundancy in the content, which doesn't help the user when trying to find content and they can find it in two or more places.

File: docs/reference/deprecated_api_list.rst [hunk 10]
```diff
@@ -6,88 +6,174 @@
...
 **********************************************************************************************
 
-Several of our API functions have been flagged for deprecation. Using the following functions results in
-errors and unexpected results, so we encourage you to update your code accordingly.
+Several of our API functions have been flagged for deprecation. Using the
+following functions results in errors and unexpected results, so we encourage
+you to update your code accordingly.
 
-Context management
+Deprecated since ROCm 1.6.0
```
Comment (randyh62) [inline docs/reference/deprecated_api_list.rst]: I like the addition of the Deprecated release, but best to reverse sort this from most recent to oldest? Also, isn't there some rule about how long things can be deprecated? 

I also like the addition of the link to the feature documentation.

Comment (matyas-streamhpc) [reply docs/reference/deprecated_api_list.rst]: Thanks Randy! I reversed the order or releases.

It is a good question about how long functions needs to be deprecated.

File: docs/sphinx/_toc.yml.in (hunk 5 above)
Comment (neon60) [reply docs/sphinx/_toc.yml.in]: Kept only annotated.

File: docs/reference/hip_runtime_api/data_structures.rst (hunk 4 above)
Comment (neon60) [reply docs/reference/hip_runtime_api/data_structures.rst]: Solved as we discussed.

File: docs/sphinx/_toc.yml.in (hunk 6 above)
Comment (neon60) [reply docs/sphinx/_toc.yml.in]: Removed as we discussed.

File: docs/reference/hip_runtime_api/files.rst (hunk 9 above)
Comment (neon60) [reply docs/reference/hip_runtime_api/files.rst]: Fixed.

File: docs/index.md (hunk 1 above)
Comment (neon60) [reply docs/index.md]: This is a separate issue. Let's discuss later.

File: docs/reference/hip_runtime_api_reference.rst (hunk 2 above)
Comment (neon60) [reply docs/reference/hip_runtime_api_reference.rst]: Solved as we discussed.

File: docs/reference/hip_runtime_api/modules_list.rst (hunk 3 above)
Comment (neon60) [reply docs/reference/hip_runtime_api/modules_list.rst]: We need to keep it to be able to link from deprecated list.

File: docs/sphinx/_toc.yml.in (hunk 7 above)
Comment (neon60) [reply docs/sphinx/_toc.yml.in]: Removed as we discussed.

File: docs/reference/hip_runtime_api/modules_list.rst (hunk 8 above)
Comment (neon60) [reply docs/reference/hip_runtime_api/modules_list.rst]: Updated.

**Review by randyh62 — APPROVED** [review]

**Review by jujiang-del — APPROVED** [review]
Looks good to me
